### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by davmillar**

### CODE REVIEW

File: packages/@honkit/markdown/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
   "dependencies": {
     "@honkit/html": "workspace:*",
     "@honkit/markup-it": "workspace:*",
-    "markdown-escape": "^1.1.0"
+    "markdown-escape": "^2.0.0"
```
Comment (azu) [inline packages/@honkit/markdown/package.json:43]: https://github.com/kemitchell/markdown-escape.js/compare/v1.1.0...v2.0.0

File: packages/@honkit/markup-it/package.json [hunk 2]
```diff
@@ -34,22 +34,21 @@
   "dependencies": {
     "escape-string-regexp": "^4.0.0",
     "html-entities": "^1.4.0",
-    "htmlparser2": "^4.1.0",
+    "htmlparser2": "^6.0.0",
```
Comment (azu) [inline packages/@honkit/markup-it/package.json:37]: https://github.com/fb55/htmlparser2/releases/tag/v5.0.0
https://github.com/fb55/htmlparser2/releases/tag/v6.0.0

File: packages/@honkit/markup-it/package.json [hunk 3]
```diff
@@ -34,22 +34,21 @@
...
   },
   "devDependencies": {
 		"@eslint/js": "^9.1.1",
     "@stylistic/eslint-plugin-js": "^2.8.0",
     "@types/jest": "^29.5.13",
     "@types/node": "^22.7.4",
-    "eslint": "^9.1.1",
-		"globals": "^15.0.0",
+    "eslint": "^9.11.1",
+		"globals": "^15.10.0",
```
Comment (azu) [inline packages/@honkit/markup-it/package.json:48]: indent is a little strange.

File: packages/@honkit/markup-it/package.json [hunk 4]
```diff
@@ -34,22 +34,21 @@
...
     "jest": "^29.7.0",
-    "jsdom": "^20.0.3",
-    "rimraf": "^3.0.2",
+    "jsdom": "^23.0.0",
```
Comment (azu) [inline packages/@honkit/markup-it/package.json:50]: https://github.com/jsdom/jsdom/releases/tag/22.0.0
https://github.com/jsdom/jsdom/releases/tag/23.0.0

File: packages/honkit/package.json [hunk 5]
```diff
@@ -45,19 +45,19 @@
     "bash-color": "^0.0.4",
     "cheerio": "^1.0.0",
     "chokidar": "^3.6.0",
-    "commander": "^5.1.0",
+    "commander": "^6.2.1",
```
Comment (azu) [inline packages/honkit/package.json:48]: https://github.com/tj/commander.js/releases/tag/v6.0.0

File: packages/honkit/package.json [hunk 6]
```diff
@@ -45,19 +45,19 @@
...
     "cp": "^0.2.0",
     "cpr": "^3.0.1",
-    "crc": "^3.8.0",
+    "crc": "^4.3.2",
```
Comment (azu) [inline packages/honkit/package.json:51]: https://github.com/alexgorbatchev/crc/releases/tag/v4.0.0

File: packages/honkit/package.json [hunk 7]
```diff
@@ -45,19 +45,19 @@
...
     "destroy": "^1.2.0",
     "direction": "^0.1.5",
-    "dom-serializer": "^0.1.1",
-    "error": "7.0.2",
+    "dom-serializer": "^2.0.0",
```
Comment (azu) [inline packages/honkit/package.json:54]: https://github.com/cheeriojs/dom-serializer/releases/tag/v2.0.0

File: packages/honkit/package.json [hunk 8]
```diff
@@ -45,19 +45,19 @@
...
+    "error": "^7.0.2",
     "escape-html": "^1.0.3",
     "escape-string-regexp": "^4.0.0",
     "extend": "^3.0.2",
-    "flat-cache": "^2.0.1",
-    "front-matter": "^2.3.0",
+    "flat-cache": "^5.0.0",
+    "front-matter": "^4.0.2",
```
Comment (azu) [inline packages/honkit/package.json:60]: https://github.com/jaredwray/flat-cache/releases/tag/v4.0.0
https://github.com/jaredwray/flat-cache/releases/tag/v5.0.0

Comment (azu) [inline packages/honkit/package.json:60]: https://github.com/jxson/front-matter?tab=readme-ov-file#300

File: packages/honkit/package.json [hunk 9]
```diff
@@ -70,10 +70,10 @@
     "js-yaml": "^3.14.1",
     "json-schema-defaults": "^0.1.1",
     "jsonschema": "1.1.0",
-    "juice": "^8.1.0",
+    "juice": "^9.0.1",
```
Comment (azu) [inline packages/honkit/package.json:73]: https://github.com/Automattic/juice/releases/tag/v9.0.0

File: packages/honkit/package.json [hunk 10]
```diff
@@ -83,9 +83,9 @@
     "q": "^1.5.1",
     "resolve": "^1.22.8",
     "semver": "^7.6.3",
-    "send": "^0.17.2",
-    "tiny-lr": "^1.1.1",
-    "tmp": "0.0.28",
+    "send": "^1.1.0",
```
Comment (azu) [inline packages/honkit/package.json:86]: https://github.com/pillarjs/send/releases/tag/1.1.0

**Review by azu — COMMENTED** [review]
I have a few questions.

1

What criteria do you use for updating?
Some of them don't seem to be the latest versions.


2

How did you test them?
The build was successful and the output seems to be the same (snapshot test https://github.com/honkit/honkit/blob/master/packages/honkit/src/__tests__/snapshot-honkit.ts).
I'm a little concerned that some features are broken due to the wide range of impact.
Live Reload (`honkit serve`) is not automatically tested, so we need to manually check if it works.

**Comment by davmillar** [comment]
I've done a rebase off of the current master branch and merged and cleaned up some of the commits. I'll do some additional testing tomorrow but have pushed it to see how automated tests perform and get any other feedback if folks wanna take a cursory examination.

**Comment by davmillar** [comment]
I'm gonna retract this and tackle things one at a time in a new branch. Sorry for wasting your time @azu and thanks for your patience and feedback!

**Pull request closed without merging**